File: 1962-remove-stones-to-minimize-the-total/1962-remove-stones-to-minimize-the-total.cpp
class Solution {
public:
    int minStoneSum(vector<int>& piles, int k) {
        priority_queue<int>pq(piles.begin(),piles.end());
        int sum= accumulate(piles.begin(),piles.end(),0);
        while(k--){
            int val = pq.top();
            pq.pop();
            pq.push(val-val/2);
            sum-=val/2;
        }
        return sum;
    }
};

File: 1962-remove-stones-to-minimize-the-total/1962-remove-stones-to-minimize-the-total.java
class Solution {
    public int minStoneSum(int[] piles, int k) {
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        int sum=0;
        int n = piles.length;
        for(int i=0; i<n; i++){
            pq.add(piles[i]);
        }
        while(k-->0){
            int val = pq.remove();
            int newval =(int) Math.ceil((val*1.0)/2);
            // System.out.print("val "+val+"newval "+newval+" ");
            pq.add(newval);
        }
        while(pq.size()>0){
            sum+=pq.remove();
        }
        return sum;
    }
}